Common/SGD/src/main/java/org/tribuo/common/sgd/AbstractSGDTrainer.java

```diff
@@ -158,17 +161,23 @@ public V train(Dataset<T> examples, Map<String, Provenance> runProvenance) {
         U[] sgdTargets = (U[]) new Object[examples.size()];
         double[] weights = new double[examples.size()];
         int n = 0;
+        long featureSize = 0;
+        long denseCount = 0;
         for (Example<T> example : examples) {
             weights[n] = example.getWeight();
             if (example.size() == featureSpaceSize) {
                 sgdFeatures[n] = DenseVector.createDenseVector(example, featureIDMap, addBias);
+                denseCount++;
             } else {
                 sgdFeatures[n] = SparseVector.createSparseVector(example, featureIDMap, addBias);
             }
             sgdTargets[n] = getTarget(outputIDInfo,example.getOutput());
+            featureSize += sgdFeatures[n].numActiveElements();
             n++;
         }
         logger.info(String.format("Training SGD model with %d examples", n));
+        logger.fine("Average feature size = " + featureSize / (double)n);
+        logger.fine("Dense count = " + denseCount);
```

Comment:
These log messages aren't clear to someone who hasn't read this code. Maybe something like: 
```suggestion
        logger.fine("Mean number of active features = " + featureSize / (double)n);
        logger.fine("Number of dense examples = " + denseCount);
```

Reply:
Sure. To be honest this could go, I was using it to track down a performance issue, but it'll probably be useful logging for the next performance issue.